### ISSUE TITLE
Cannot install SocketIOClient with its dependencies

### ISSUE THREAD

**Issue by Oneiros90** [GlitchEnzo/NuGetForUnity#426]
## Description
Hi, it looks like I cannot install SocketIOClient successfully with N4U.
When I click "install" on it, after the compilation is done, I see this error:

```
Assembly 'Assets/Packages/System.Text.Json.5.0.2/lib/netstandard2.0/System.Text.Json.dll' will not be loaded due to errors:
Unable to resolve reference 'System.Runtime.CompilerServices.Unsafe'. Is the assembly missing or incompatible with the current platform?
Reference validation can be disabled in the Plugin Inspector.
```

If I try to open the dependency tree (`NuGet/Show Dependency Tree`) these are the errors I see:
```
System.Collections 4.3.0 is not installed!
System.Buffers 4.5.1 is not installed!
System.Memory 4.5.4 is not installed!
System.Numerics.Vectors 4.5.0 is not installed!
System.Runtime.CompilerServices.Unsafe 5.0.0 is not installed!
System.Threading.Tasks.Extensions 4.5.4 is not installed!
```

I tried to manually install these packages through N4U but it didn't work, it looks like it considers these packages as already included in the engine. I found a workaround by commenting out the parts where the [IsAlreadyImportedInEngine](https://github.com/GlitchEnzo/NuGetForUnity/blob/16406e75f625b71df662e858508d0ea60da565ff/Assets/NuGet/Editor/NugetHelper.cs#L553) function is used. The missing packages are correctly downloaded this way and the SocketIOClient seems to work fine in playmode (I didn't try yet in a builded version).

I'm sure my workaround was not the right solution, I would like to know if there's a proper way to deal with these dependencies. 
Thanks!

- **NuGet Package:** [SocketIOClient 3.0.3](https://www.nuget.org/packages/SocketIOClient/)
- **NuGetForUnity Version:** 3.0.2 (from source code, not from released package)
- **Unity Version:** Unity 2021.2.0f1
- **Operating System:** Windows 10 (same issue on macOS)

**Comment by WildlandHerald** [comment]
Can you please tell what steps did you perform to compile the extension? I'm struggling with that, `build.ps1` spits out bunch of errors if I try to run it without opening the project in unity, and if I do open it in unity it complains about collision between Newtonsoft.JSON in Library/PackageCache and Netwonsoft.JSON in Assets/Packages.

**Comment by Oneiros90** [comment]
> Can you please tell what steps did you perform to compile the extension? I'm struggling with that, `build.ps1` spits out bunch of errors if I try to run it without opening the project in unity, and if I do open it in unity it complains about collision between Newtonsoft.JSON in Library/PackageCache and Netwonsoft.JSON in Assets/Packages.

I'm sorry, are you sure this is related to my issue? I was not having problems with `Newtonsoft.JSON` and I don't know what's `build.ps1`

**Comment by WildlandHerald** [comment]
> > Can you please tell what steps did you perform to compile the extension? I'm struggling with that, `build.ps1` spits out bunch of errors if I try to run it without opening the project in unity, and if I do open it in unity it complains about collision between Newtonsoft.JSON in Library/PackageCache and Netwonsoft.JSON in Assets/Packages.
> 
> I'm sorry, are you sure this is related to my issue? I was not having problems with `Newtonsoft.JSON` and I don't know what's `build.ps1`

Oh, so you didn't build the NuGetForUnity package with your changes? How did you make it work without building?

**Comment by JoC0de** [comment]
I checked it is fixed by #431.
